prereise/gather/hiflddata/data_trans.py

```diff
@@ -0,0 +1,433 @@
+    for i in range(len(zone)):
+        zone_dic[zone['STATE'][i]]=zone['ID'][i]
+    return zone_dic
+
+
+# Clean data; throw substations which are outside the United States and not available.
+def Clean(E_csv, zone_dic):
+    csv_data = pd.read_csv(E_csv)
+    Num_sub = len(csv_data)
+    row_indexs = []
+    for i in range(Num_sub):
+        if((csv_data['STATE'][i] not in zone_dic) or (csv_data['STATUS'][i] != 'IN SERVICE') or (csv_data['LINES'][i] == 0)):
+            row_indexs.append(i)
+    clean_data = csv_data.drop(labels = row_indexs)
+    return clean_data
+
+
+
+#Create dictionary to store neighbors of each sub
+def Neighbors(sub_by_coord_dict, sub_name_dict, raw_lines):
+    N_dict = {}
+    nodes = []
+    lines = []
+    missinglines = []
+    if not os.path.isfile('data/unzip/Electric_Power_Transmission_Lines.geojson'):
+        with zipfile.ZipFile('data/Electric_Power_Transmission_Lines.geojson.zip', 'r') as zip_ref:
+            zip_ref.extractall('data/unzip')
+    with open('data/unzip/Electric_Power_Transmission_Lines.geojson','r',encoding='utf8') as fp:
+        data = json.load(fp)
+    for i in range(len(data['features'])):
+        line = data['features'][i]
+        line_id = line['properties']['ID']
+        line_type = line['properties']['TYPE'] # e.g. "AC; OVERHEAD"
+        if line['properties']['SUB_1'] == 'NOT AVAILABLE' or line['properties']['SUB_2'] == 'NOT AVAILABLE':
+            # quite a few of NOT AVAILABLE scenarios in transmission rawdata
+            print("INFO: sub1 or sub2 NOT AVAILABLE for transmission line: ", line)
+            missinglines.append(line)
+            continue
+
+        if line['properties']['SUB_1'] not in sub_name_dict or line['properties']['SUB_2'] not in sub_name_dict:
+            # e.g., ID 201978 with SUB2 UNKNOWN202219, never shown in substation rawdata
+            print("INFO: sub1 or sub2 name not identified for transmission line: ", line)
+            missinglines.append(line)
+            continue
+
+        start_coord = (line['geometry']['coordinates'][0][0][1], line['geometry']['coordinates'][0][0][0])
+        end_coord = (line['geometry']['coordinates'][0][-1][1], line['geometry']['coordinates'][0][-1][0])
+        candidate_coordinates = sub_name_dict.get(line['properties']['SUB_1']) + sub_name_dict.get(line['properties']['SUB_2'])
+
+        sub1 = min(candidate_coordinates, key=lambda p: computeGeoDist(p, start_coord))
+        sub2 = min(candidate_coordinates, key=lambda p: computeGeoDist(p, end_coord))
+
+        # dist = meter2Mile(raw_lines['SHAPE_Length'][line_id])
+        dist = computeGeoDist(start_coord, end_coord)
+        # vol = raw_lines['VOLTAGE'][line_id]
+        # reactance = computeReactance(line_type, dist, vol)
+        # rateA = computeRateA(line_type, vol)
+        lines.append([line_id, line_type, sub_by_coord_dict.get(sub1)[0], sub_by_coord_dict.get(sub1)[1], sub_by_coord_dict.get(sub2)[0], sub_by_coord_dict.get(sub2)[1], dist])
```

Comment:
Code should be formatted using `black`, which does not allow lines >88 characters in length.